neo4j/notifications/notifications.go

```diff
@@ -30,42 +39,199 @@ const (
-type NotificationMinimumSeverityLevel string
+// DisabledClassifications returns the classifications that are disabled.
+func (n *NotificationDisabledClassifications) DisabledClassifications() []NotificationClassification {
+	return n.classifications
+}
 
-const (
-	DefaultLevel     NotificationMinimumSeverityLevel = ""
-	DisabledLevel    NotificationMinimumSeverityLevel = "OFF"
-	WarningLevel     NotificationMinimumSeverityLevel = "WARNING"
-	InformationLevel NotificationMinimumSeverityLevel = "INFORMATION"
-)
+// ToNotification returns a db.Notification that corresponds to the given db.GqlStatusObject.
+// It maps fields from the status to their respective notification fields.
+func ToNotification(gqlStatusObject db.GqlStatusObject) *db.Notification {
+	return &db.Notification{
+		//lint:ignore SA1019 Code is supported at least until 6.0
+		Code: gqlStatusObject.Code,
+		//lint:ignore SA1019 Title is supported at least until 6.0
+		Title:       gqlStatusObject.Title,
+		Description: gqlStatusObject.StatusDescription,
+		Position:    gqlStatusObject.Position,
+		Severity:    gqlStatusObject.Severity,
+		Category:    gqlStatusObject.Classification,
+	}
+}
+
+func newDefaultDiagnosticRecord() map[string]any {
+	return map[string]any{
+		"OPERATION":      "",
+		"OPERATION_CODE": "0",
+		"CURRENT_SCHEMA": "/",
+	}
+}
+
+func newSuccessGqlStatusObject() *db.GqlStatusObject {
+	return &db.GqlStatusObject{
+		GqlStatus:         "00000",
+		StatusDescription: "note: successful completion",
+		DiagnosticRecord:  newDefaultDiagnosticRecord(),
+	}
+}
+
+func newNoDataGqlStatusObject() *db.GqlStatusObject {
+	return &db.GqlStatusObject{
+		GqlStatus:         "02000",
+		StatusDescription: "note: no data",
+		DiagnosticRecord:  newDefaultDiagnosticRecord(),
+	}
+}
+
+func newOmittedResultGqlStatusObject() *db.GqlStatusObject {
+	return &db.GqlStatusObject{
+		GqlStatus:         "00001",
+		StatusDescription: "note: successful completion - omitted result",
+		DiagnosticRecord:  newDefaultDiagnosticRecord(),
+	}
+}
+
+func newUnknownWarningResultGqlStatusObject() *db.GqlStatusObject {
+	return &db.GqlStatusObject{
+		GqlStatus:         "01N42",
+		StatusDescription: "warn: unknown warning",
+		DiagnosticRecord:  newDefaultDiagnosticRecord(),
+	}
+}
+
+func newUnknownInformationResultGqlStatusObject() *db.GqlStatusObject {
+	return &db.GqlStatusObject{
+		GqlStatus:         "03N42",
+		StatusDescription: "info: unknown notification",
+		DiagnosticRecord:  newDefaultDiagnosticRecord(),
+	}
+}
+
+// ToGqlStatusObject returns a db.GqlStatusObject that corresponds to the given db.Notification.
+// It maps fields from the notification to their respective status fields.
+func ToGqlStatusObject(notification db.Notification) *db.GqlStatusObject {
+	defaultStatus := newUnknownInformationResultGqlStatusObject()
+	if notification.Severity == string(WarningLevel) {
+		defaultStatus = newUnknownWarningResultGqlStatusObject()
+	}
```

Comment:
```suggestion
	var defaultStatus *db.GqlStatusObject
	if notification.Severity == string(WarningLevel) {
		defaultStatus = newUnknownWarningResultGqlStatusObject()
	} else {
		defaultStatus = newUnknownInformationResultGqlStatusObject()
	}
```
Avoid potentially wasted allocations.